src/AWS.Deploy.Recipes/RecipeDefinitions/ASP.NETAppElasticBeanstalk.recipe

```diff
@@ -139,11 +140,67 @@
             "Id": "InstanceType",
             "Name": "EC2 Instance Type",
             "Description": "The EC2 instance type of the EC2 instances created for the environment.",
-            "Type": "String",
+            "Type": "Object",
             "TypeHint": "InstanceType",
-            "DefaultValue": "",
             "AdvancedSetting": true,
-            "Updatable": true
+            "Updatable": true,
+            "ChildOptionSettings": [
+                {
+                    "Id": "InstanceFreeTierEligible",
+                    "Name": "Free Tier Eligible",
+                    "Description": "Do you want the EC2 instance to be free tier eligible?",
+                    "Type": "Bool",
+                    "DefaultValue": true,
+                    "AdvancedSetting": true,
+                    "Updatable": false
+                },
+                {
+                    "Id": "InstanceArchitecture",
+                    "Name": "EC2 Instance Architecture",
+                    "Description": "The architecture of the EC2 instances created for the environment.",
+                    "Type": "String",
+                    "DefaultValue": "x86_64",
+                    "AllowedValues": [
+                        "x86_64",
+                        "arm64"
+                    ],
+                    "AdvancedSetting": true,
+                    "Updatable": true
+                },
+                {
+                    "Id": "InstanceCPUCores",
+                    "ParentSettingId": "InstanceType",
+                    "Name": "EC2 Instance CPU Cores",
+                    "Description": "The CPU cores of the EC2 instances created for the environment.",
+                    "Type": "Int",
```

Comment:
Can you add a range validator for the setting and add a validator unit test?

Reply:
The CPU cores type is dynamically calculated based on the type hint. It is selected from a range of values presented to the end users. It is also not a real option setting item, it is only used to filter the INstance Type. There is no predefined list of values to set as a validator.

Reply:
Currently, the type hints are not invoked if a user choses to perform a deployment via a configuration file. It is possible for a user to supply an arbitrarily large value or negative values in the config. Having a simple range validator will throw an actionable error before the CDK hand-off happens.

Reply:
even if the user sets an invalid valid, this does not affect anything. These are filtering OSI, not actual OSI. They don't have a predetermined value, so we cannot create a random range to validate.